benches/benches/block_target_gas.rs

```diff
@@ -345,6 +381,97 @@ fn service_with_contract_id(
     (service, rt)
 }
 
+fn service_with_many_contracts(
+    state_size: u64,
+    contract_ids: Vec<ContractId>,
+) -> (FuelService, tokio::runtime::Runtime) {
+    use fuel_core::database::vm_database::IncreaseStorageKey;
+    let rt = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap();
+    let _drop = rt.enter();
+    let mut database = Database::rocksdb();
+    let mut config = Config::local_node();
+    config
+        .chain_conf
+        .consensus_parameters
+        .tx_params
+        .max_gas_per_tx = TARGET_BLOCK_GAS_LIMIT;
+
+    let contract_configs = contract_ids
+        .iter()
+        .map(|contract_id| ContractConfig {
+            contract_id: *contract_id,
+            code: vec![],
+            salt: Default::default(),
+            state: None,
+            balances: None,
+            tx_id: None,
+            output_index: None,
+            tx_pointer_block_height: None,
+            tx_pointer_tx_idx: None,
+        })
+        .collect::<Vec<_>>();
+    config.chain_conf.initial_state.as_mut().unwrap().contracts = Some(contract_configs);
+
+    config
+        .chain_conf
+        .consensus_parameters
+        .predicate_params
+        .max_gas_per_predicate = TARGET_BLOCK_GAS_LIMIT;
+    config.chain_conf.block_gas_limit = TARGET_BLOCK_GAS_LIMIT;
+    config.chain_conf.consensus_parameters.gas_costs = GasCosts::new(default_gas_costs());
+    config
+        .chain_conf
+        .consensus_parameters
+        .fee_params
+        .gas_per_byte = 0;
+    config.utxo_validation = false;
+    config.block_production = Trigger::Instant;
+
+    let mut storage_key = primitive_types::U256::zero();
+    let mut key_bytes = Bytes32::zeroed();
+
+    for contract_id in contract_ids.iter() {
```

Comment:
Initializing state and balances for all contracts will kill us=D It's okay to init only for `ContractId::zeroed()`.

Reply:
How will it kill us? It works. I'm not doing full state like for the other tests--I can still reduce it some as well. But I need separate contract ids for each input/output. Each contract needs to exist in the DB as well.

Reply:
10M state for 254 contracts will take a lot of time to initialize=) We only need to init state for the `ContractId::zeroed()`. You can simply deploy other contracts

Reply:
Like I said,  I'm not doing the full 10M state for all of them:
```rs
    pub fn new_with_many_contracts(
        contract_ids: Vec<ContractId>,
    ) -> MultiContractBenchmarkRunnerBuilder {
        let state_size = 1000; // Arbitrary small state size
        let (service, rt) = service_with_many_contracts(state_size, contract_ids.clone());
        let rng = rand::rngs::StdRng::seed_from_u64(2322u64);
        MultiContractBenchmarkRunnerBuilder {
            service,
            rt,
            contract_ids,
            rng,
        }
    }
```

as opposed to what we're doing for the shared contract:
```rs
    pub fn new_shared(contract_id: ContractId) -> SharedSanityBenchmarkRunnerBuilder {
        let state_size = crate::utils::get_state_size();
        let (service, rt) = service_with_contract_id(state_size, contract_id);
        let rng = rand::rngs::StdRng::seed_from_u64(2322u64);
        SharedSanityBenchmarkRunnerBuilder {
            service,
            rt,
            contract_id,
            rng,
        }
    }
```

We could restructure to specify the state size for each contract though if we wanted. I don't think it's hurting any of our benchmarks significantly, currently though.